soco/zgs.py

```diff
@@ -0,0 +1,267 @@
+"""Provides handling for ZoneGroupState information.
+
+ZoneGroupState XML payloads are received from both:
+* zoneGroupTopology.GetZoneGroupState()['ZoneGroupState']
+* zoneGroupTopology subscription event callbacks
+
+The ZoneGroupState payloads are identical between all speakers in a
+household, but may be generated with differing orders for contained
+ZoneGroup or ZoneGroupMember elements and children. To benefit from
+similar contents, payloads are passed through an XSL transformation
+to normalize the data, to allow simple equality comparisons, and to
+avoid unnecessary reprocessing of identical data.
+
+Since the payloads are identical between all speakers, we can use a
+common cache per household.
+
+Satellite devices can sometimes deliver outdated payloads when they
+are directly polled. These requests are instead redirected to their
+parent device.
+
+Example payload contents:
```

Comment:
Instead of having an example payload here, perhaps it would be better to have this as a fixture file and add tests that run on that? That would also be better for maintainability (just in case the structure changes at some point).

Reply:
Yes, I agree.

Reply:
I forgot, there are already tests in place for ZGS parsing: https://github.com/SoCo/SoCo/blob/48e3b705d3ec05acd993de33aacda3ce8f96349e/tests/test_core.py#L162-L166